indexing-service/src/main/java/org/apache/druid/indexing/overlord/SingleTaskBackgroundRunner.java

```diff
@@ -178,17 +178,16 @@ public void stop()
     if (runningItem != null) {
       final Task task = runningItem.getTask();
       final long start = System.currentTimeMillis();
-      final boolean graceful;
       final long elapsed;
       boolean error = false;
 
-      if (taskConfig.isRestoreTasksOnRestart() && task.canRestore()) {
-        // Attempt graceful shutdown.
-        graceful = true;
-        log.info("Starting graceful shutdown of task[%s].", task.getId());
+      // stopGracefully for resource cleaning, independent of the fact whether the task is restorable or not
```

Comment:
nit: can delete the "independent of the fact" portion and the later "Attempt graceful shutdown" comment